### ISSUE TITLE
Missing totp 2fa in version 1.15

### ISSUE THREAD

**Issue by markusglaetzner** [kasmtech/workspaces-issues#518]
There is no totp menu in two-factor-setup (fresh installation of kasm_release_1.15.0.06fdc8 on a debian 12 vm), though the option "allow_totp_2fa" is enabled in settings for all users. I only have the possibility to use webauthn.

**Comment by j-travis** [comment]
Can you please try deleting the `allow_totp_2fa` setting from the All Users group and then re-adding it.
Hopefully that should correct the issue

**Comment by uniquegch** [comment]
yes I have the same issue that I only have the webauthn, in my 1.14.0 which was upgrade from 1.13.0 I can use the Google authentication

**Comment by uniquegch** [comment]
> yes I have the same issue that I only have the webauthn, in my 1.14.0 which was upgrade from 1.13.0 I can use the Google authentication

whereto did you move that issue?

**Comment by uniquegch** [comment]
> Can you please try deleting the `allow_totp_2fa` setting from the All Users group and then re-adding it. Hopefully that should correct the issue
in 1.14.0 it is actually called "enable_totp_two_factor"

**Comment by j-travis** [comment]
> yes I have the same issue that I only have the webauthn, in my 1.14.0 which was upgrade from 1.13.0 I can use the Google authentication

This is no direct support for webauthn in 1.14.  Its only TOTP.   In either case, both of these options only apply to local or LDAP based auth. If you are using SAML or OIDC then these settings are ignored since the MFA is handled by the identity provider

**Comment by uniquegch** [comment]
No I meant that in verson 1.14.0 it is called "enable_totp_two_factor" while on my new test setup in 1.15.0 it is "allow_totp_2fa". In 1.15.0 I did remove it and re-added it and it was working. only the QR code seems to be not so easy to scan with my app. I took a screenshot and paste it into a paint program and that way I could scan the QR. could that be made a big bigger maybe. because when I setup 2FA for SSH in the CLi it is huge and always reliable to scan.

**Comment by mathiasringhof** [comment]
In the default install "allow_totp_2fa" is of type string. When you remove and add "allow_totp_2fa" it's now of type boolean and setting it to true gets you the TOTP 2fa option again.

**Comment by hellfyred** [comment]
Confirmed @mathiasringhof's fix works on a fresh KASM 1.15.0 install.

**Comment by FDrebin** [comment]
Removing it and re-adding it resolved this issue for me. Very weird problem on a fresh install.

**Comment by teja-kasm** [comment]
We’re currently enhancing our GitHub issue tracker to improve organization and efficiency.

As part of this process, we’re closing older issues to help keep the tracker focused on current and active topics.

If this issue is still relevant or you’re still experiencing the problem, please feel free to create a new issue.

Thank you for your understanding!
